File: apiwrapper/src/main/java/com/pinkertone/apiwrapper/ApiService.java
package com.pinkertone.apiwrapper;


import okhttp3.OkHttpClient;
import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;

class ApiService {
    private final String baseUrl;
    private String authToken = "";
    private final String language;

    public IApiEndpoints apiService;

    public ApiService(String baseUrl, String authToken, String language) {
        this(baseUrl, language);
        this.authToken = authToken;
    }

    public ApiService(String baseUrl, String language) {
        this.baseUrl = baseUrl;
        this.language = language;
        initApiService();
    }

    // TODO: test with android
    private void initApiService() {

        OkHttpClient httpClient = new OkHttpClient.Builder().addInterceptor(new HeadersInterceptor(authToken, language)).build();

        Retrofit retrofit = new Retrofit.Builder()
                .baseUrl(baseUrl)
                .addConverterFactory(GsonConverterFactory.create())
                .client(httpClient)
                .build();

        apiService = retrofit.create(IApiEndpoints.class);
    }

    public void setAuthToken(String authToken) {
        this.authToken = authToken;
        initApiService();
    }

    public String getAuthToken() {
        return authToken;
    }
}

File: apiwrapper/src/main/java/com/pinkertone/apiwrapper/ApiService.kt
package com.pinkertone.apiwrapper

import okhttp3.OkHttpClient
import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory

class ApiService constructor(
    private val baseUrl: String,
    private val language: String,
) {
    private var authToken: String = ""

    lateinit var api: IApiEndpoints

    constructor(
        baseUrl: String,
        authToken: String,
        language: String
    ) : this(baseUrl, language) {
        this.authToken = authToken
    }

    init {
        initApiService()
    }

    fun setAuthToken(token: String?) {
        this.authToken = token ?: ""
        initApiService()
    }

    fun getAuthToken(): String {
        return authToken
    }

    // TODO: test with android
    private fun initApiService() {
        val httpClient =
            OkHttpClient.Builder().addInterceptor(HeadersInterceptor(authToken ?: "", language))
                .build()
        val retrofit = Retrofit.Builder()
            .baseUrl(baseUrl)
            .addConverterFactory(GsonConverterFactory.create())
            .client(httpClient)
            .build()
        api = retrofit.create(IApiEndpoints::class.java)
    }
}